sample-3.md

```diff
@@ -0,0 +1,537 @@
+
+This step-by-step guide will guide you through all the steps involved to 
+run “Hello World” eventing solution, where you will send an event to the 
+Knative eventing broker inside your K8s namespace and receive it in a pod 
+using Knative eventing. ETA - 20-30 mins. If you want to skip all the steps 
+and want to quickly deploy the solution then please refer to 
+[Quick Start: Deploy "Hello World" eventing solution](todo).
+
+## Before you begin
+
+Before you begin:
+- Have a Kubernetes Cluster running version 1.12.
+- [Install Knative Eventing](todo) and verify the health of the Knative Eventing
+  control plane.
+  
+## Description
+
+This sample is identical to [Quick Start: Deploy "Hello World" eventing solution](todo),
+but rather than using a single YAML, we will apply each piece one step at a time, 
+explaining what it does as we do so.
+
+We will create the following topology:
+
+Image from 2.
+
+- [`Pod` named `curl`](#event-producer), acting as the event producer.
+- `Broker` named `default`.
+- Two [`Trigger`s](#triggers), each with a mutually exclusivefilter pointing at distinct 
+  subscribers.
+- Two Kubernetes [`Service`s](#event-consumers), each pointing at a unique:
+    - [`Deployment`](#event-consumers), acting as the event consumer.
+
+## Helpers
+
+First we will create useful shell variables to allow us to not have to type 
+things out every time.
+
+```shell
+K_NAMESPACE=kn-eventing-step-by-step-sample
+```
+
+## Namespace
+
+Create the namespace.
+
+```shell
+kubectl create namespace $K_NAMESPACE
+```
+
+Now that the namespace has been created, we will label it to get Knative eventing
+setup. Adding this label will:
+- Create `ServiceAccount`s.
+- Create `RoleBinding`s to give permissions to those `ServiceAccount`s.
+- Create a `Broker` named `default`.
+
+```shell
+kubectl label namespace $K_NAMESPACE knative-eventing-injection=enabled
+```
+
+### Broker
+
+What is a `Broker`? It can be thought of as an event mesh. Event
+producers send events into the `Broker`. Event consumers register the kinds of events
+they are interested in with the `Broker` (via `Trigger`s, described later). The
+`Broker` ensures that every event sent to the `Broker` by event producers is sent
+to all interested event consumers.
+
+
+Let's verify that the `Broker` is in a healthy state.
+
+```shell
+kubectl -n $K_NAMESPACE get broker default
+```
+
+This should return something like:
+
+```shell
+NAME      READY   REASON   HOSTNAME                                                           AGE
+default   True             default-broker.kn-eventing-step-by-step-sample.svc.cluster.local   1m
+```
+
+We are particularly interested in the second column, `READY`. We want that value to
+be `True`. It may take a minute or two to become `True`. If it is not `True` within
+two minutes, then something has gone wrong. Hopefully the `REASON` column gives a
+useful error message. See our [Broker Debugging Guide](somewhere) to help fix the
```

Comment:
Add it to the docs. I am thinking of creating issues for whatever we wouldn't be able to write today and can also ask wider community (over slack) to pick these issues.

Reply:
Added to the shared spreadsheet.